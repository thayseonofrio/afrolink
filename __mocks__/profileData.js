module.exports = {
	city: "Porto Alegre",
	country: "Brasil",
	experience: 12,
	gender: "Feminino",
	jobTitle: ["Desenvolvedora"],
	name: "Teste",
	skills: ["MongoDB", "JavaScript"],
	socialLinks: {email: "test@hotmail.com"},
	state: "RS"
}