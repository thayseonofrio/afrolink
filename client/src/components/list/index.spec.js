import React from "react"
import apis from "../../services/profile"
import List from "."
import { render } from "@testing-library/react"
import profileData from "../../../../__mocks__/profileData"

jest.mock("../../services/profile")

describe("<List />", () => {
	it("should render list with profile", async () => {
		const response = {data: [profileData]}
		apis.getAllProfiles.mockImplementationOnce(() => Promise.resolve(response))
		const { container, findByText} = render(<List />)
		expect( await findByText("Desenvolvedora")).toBeInTheDocument()
		expect( await findByText("Feminino")).toBeInTheDocument()
		expect( await findByText("Teste")).toBeInTheDocument()
		expect(container).toMatchSnapshot()
	})
})