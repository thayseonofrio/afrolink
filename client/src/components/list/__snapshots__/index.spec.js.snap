// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render list with profile 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="makeStyles-root-3"
          >
            <h2
              class="makeStyles-title-5"
            >
               
              Teste
               
            </h2>
            <div
              class="makeStyles-info-4"
            >
              <span>
                Feminino
              </span>
              <span>
                 
                Desenvolvedora
                 
              </span>
              <span>
                12
                 meses 
              </span>
              <span>
                Porto Alegre
                /
                RS
              </span>
              <div
                class="makeStyles-skills-6"
              >
                <span
                  class="label"
                >
                   
                  MongoDB
                   
                </span>
                <span
                  class="label"
                >
                   
                  JavaScript
                   
                </span>
              </div>
            </div>
            <hr
              class="makeStyles-line-7"
            />
            <div
              class="makeStyles-socialLinks-8"
            >
              <a
                class="social"
                href="mailto:test@hotmail.com"
                target="_blank"
              >
                <img
                  alt="email link"
                  src="test-file-stub"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
